src/Streamly/Internal/Data/Fold.hs

```diff
@@ -1042,9 +1042,9 @@ splitAt n fld = serialWith (,) (take n fld)
 -- Stops - when the predicate succeeds.
 --
 -- /Pre-release/
-{-# INLINE sliceSepBy #-}
-sliceSepBy :: Monad m => (a -> Bool) -> Fold m a b -> Fold m a b
-sliceSepBy predicate (Fold fstep finitial fextract) =
+{-# INLINE takeEndBy_ #-}
+takeEndBy_ :: Monad m => (a -> Bool) -> Fold m a b -> Fold m a b
```

Comment:
I am not very comfortable with `takeEndBy_` and `takeEndBy`. I'd like to have `while` somewhere is the word.
I don't have better suggestions so this should work for now.